File: app/src/main/java/com/omarea/gesture/AccessibilityServiceSceneKeyEvent.java
package com.omarea.gesture;

import android.accessibilityservice.AccessibilityService;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.content.res.Configuration;
import android.view.accessibility.AccessibilityEvent;

public class AccessibilityServiceSceneKeyEvent extends AccessibilityService {
    boolean isLandscapf = false;
    private FloatVitualTouchBar floatVitualTouchBar = null;
    private BroadcastReceiver configChanged = null;

    private void hidePopupWindow() {
        if (floatVitualTouchBar != null) {
            floatVitualTouchBar.hidePopupWindow();
            floatVitualTouchBar = null;
        }
    }

    @Override
    public void onAccessibilityEvent(AccessibilityEvent event) {

    }

    @Override
    public void onServiceConnected() {
        super.onServiceConnected();

        if (configChanged == null) {
            configChanged = new BroadcastReceiver() {
                @Override
                public void onReceive(Context context, Intent intent) {
                    createPopupView();
                }
            };

            registerReceiver(configChanged, new IntentFilter(getString(R.string.action_config_changed)));
        }
        createPopupView();
    }

    @Override
    public boolean onUnbind(Intent intent) {
        hidePopupWindow();
        return super.onUnbind(intent);
    }

    @Override
    public void onInterrupt() {

    }

    // 监测屏幕旋转
    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        if (floatVitualTouchBar != null && newConfig != null) {
            if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
                isLandscapf = false;
            } else if (newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
                isLandscapf = true;
            }
            createPopupView();
        }
    }

    private void createPopupView() {
        hidePopupWindow();
        floatVitualTouchBar = new FloatVitualTouchBar(this, isLandscapf);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (floatVitualTouchBar != null) {
            floatVitualTouchBar.hidePopupWindow();
        }

        if (configChanged != null) {
            unregisterReceiver(configChanged);
            configChanged = null;
        }
    }
}

File: app/src/main/java/com/omarea/gesture/AccessibilityServiceSceneKeyEvent.kt
package com.omarea.gesture

import android.accessibilityservice.AccessibilityService
import android.content.Intent
import android.content.res.Configuration
import android.view.accessibility.AccessibilityEvent

/**
 * Created by helloklf on 2016/8/27.
 */
class AccessibilityServiceSceneKeyEvent : AccessibilityService() {
    override fun onInterrupt() {
    }
    private var floatVitualTouchBar: FloatVitualTouchBar? = null

    private fun hidePopupWindow() {
        if (floatVitualTouchBar != null) {
            floatVitualTouchBar!!.hidePopupWindow()
            floatVitualTouchBar = null
        }
    }

    override fun onServiceConnected() {
        super.onServiceConnected()
        createPopupView()
    }

    override fun onUnbind(intent: Intent?): Boolean {
        hidePopupWindow()
        return super.onUnbind(intent)
    }

    var isLandscapf = false

    // 监测屏幕旋转
    override fun onConfigurationChanged(newConfig: Configuration?) {
        super.onConfigurationChanged(newConfig)
        if (floatVitualTouchBar != null && newConfig != null) {
            if (newConfig.orientation == Configuration.ORIENTATION_PORTRAIT) {
                isLandscapf = false
            } else if (newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE) {
                isLandscapf = true
            }
            createPopupView()
        }
    }

    private fun createPopupView() {
        hidePopupWindow()
        floatVitualTouchBar = FloatVitualTouchBar(
                this,
                isLandscapf,
                true
        )
    }

    override fun onDestroy() {
        super.onDestroy()
        if (floatVitualTouchBar != null) {
            floatVitualTouchBar!!.hidePopupWindow()
        }
    }

    override fun onAccessibilityEvent(event: AccessibilityEvent) {
    }
}